src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ConstructedEETypeNode.cs

```diff
@@ -141,7 +141,10 @@ protected override DependencyList ComputeNonRelocationBasedDependencies(NodeFact
                 DefType templateType = GenericTypesTemplateMap.GetActualTemplateTypeForType(factory, _type.ConvertToCanonForm(CanonicalFormKind.Specific));
 
                 if (templateType.IsCanonicalSubtype(CanonicalFormKind.Any) && !factory.NecessaryTypeSymbol(templateType).RepresentsIndirectionCell)
+                {
                     dependencyList.Add(factory.NativeLayout.TemplateTypeLayout(templateType), "Template Type Layout");
+                    dependencyList.Add(factory.ConstructedTypeSymbol(templateType), "Template type");
```

Comment:
My preference I think would be to make the ConstructedTypeSymbol required by the TemplateTypeLayoutNode. Eventually, we will end up building dependency chains for TemplateTypeLayout, and I'd like to be able to simply add the template type layout to the tree and have it work.